### DISCUSSION TITLE
on_hover_text not possible if widget is disabled, how to explain to users why the widget is disabled?

### DISCUSSION THREAD

**Discussion by TheLeonsver1** [emilk/egui#648] [Q&A] [answered]
I'm working on a small database application.
it has a button called "New Column", the button is disabled when there are no tables in the currently open database(because there's no table to add a column to).
`on_hover_text` appears to not display when widgets are disabled and is documented to not do so ( I'm on egui 0.13.1 i think), and that means i have no easy way to tell my users why they can't click on things at times.. is there any reason why this behavior was chosen? and is there a way to configure it?

thanks!

**Comment by emilk** [comment, marked as answer]
Use [`on_disabled_hover_text`](https://docs.rs/egui/0.14.0/egui/struct.Response.html#method.on_disabled_hover_text)

**Reply by TheLeonsver1** [reply]
Thank you very much!
